pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentLongPairSet.java

```diff
@@ -190,11 +187,16 @@ public int removeIf(LongPairPredicate filter) {
      * @return a new list of keys with max provided numberOfItems (makes a copy)
      */
     public Set<LongPair> items(int numberOfItems) {
-        Set<LongPair> items = new HashSet<>();
+        return items(numberOfItems, (item1, item2) -> new LongPair(item1, item2));
+    }
+
+    @Override
+    public <T> Set<T> items(int numberOfItems, BiFunction<Long, Long, T> longPairConverter) {
```

Comment:
If we use some kind a specialized interface we can avoid 2 `long` -> `Long` conversions: 

```java
interface LongPairFunction<T> {
    T apply(long a, long b);
}

Reply:
yes, I wanted to create it custom interface to reduce number of obj creation but then I missed it. Good catch. I will fix it.